src/viam/ur/module/ur_arm.cpp

```diff
@@ -796,23 +811,50 @@ void URArm::keep_alive_() {
 void URArm::move_(std::shared_lock<std::shared_mutex> config_rlock, std::list<Eigen::VectorXd> waypoints, const std::string& unix_time) {
     auto our_config_rlock = std::move(config_rlock);
 
+    // get current joint position and add that as starting pose to waypoints
+    VIAM_SDK_LOG(info) << "move: get_joint_positions start " << unix_time;
+    auto curr_joint_pos_deg = get_joint_positions_(our_config_rlock);
+    VIAM_SDK_LOG(info) << "move: get_joint_positions end " << unix_time;
+    auto curr_joint_pos_rad = Eigen::VectorXd(curr_joint_pos_deg.size());
+    std::transform(std::make_move_iterator(curr_joint_pos_deg.begin()),
+                   std::make_move_iterator(curr_joint_pos_deg.end()),
+                   curr_joint_pos_rad.begin(),
+                   degrees_to_radians<double>);
+
+    if (reject_move_request_threshold_deg) {
+        auto delta_pos = (waypoints.front() - curr_joint_pos_rad);
+
+        if (radians_to_degrees(delta_pos.lpNorm<Eigen::Infinity>()) > *reject_move_request_threshold_deg) {
+            std::stringstream err_string;
+
+            err_string << "rejecting move request : difference between starting trajectory position [(";
+            std::transform(std::make_move_iterator(waypoints.front().cbegin()),
+                           std::make_move_iterator(waypoints.front().cend()),
+                           std::experimental::make_ostream_joiner(err_string, ", "),
```

Comment:
Use `boost/io/ostream_joiner.hpp` rather than `std::experimental` which may not be reliably implemented.